### ISSUE TITLE
Automate the updating of OmniSharp version

### ISSUE THREAD

**Issue by mthalman** [dotnet/source-build#3577]
The source-build smoke tests use a [pinned version of OmniSharp](https://github.com/dotnet/installer/pull/17074) to test with. This should be regularly maintained to refer to the latest release, ideally through automation. Once this is done, we can remove the [maintenance item from the release checklist](https://github.com/dotnet/source-build/pull/3570) as well.

**Comment by MichaelSimons** [comment]
[Triage] This issue is being superseded by https://github.com/dotnet/source-build/issues/3690.
